static/src/javascripts/projects/common/modules/weather.js

```diff
@@ -0,0 +1,236 @@
+ */
+
+define([
+    'bean',
+    'raven',
+    'common/utils/_',
+    'common/utils/$',
+    'common/utils/ajax',
+    'common/utils/config',
+    'common/utils/mediator',
+    'common/utils/template',
+    'common/modules/search-tool',
+    'common/modules/userPrefs',
+    'text!common/views/weather.html'
+], function (
+    bean,
+    raven,
+    _,
+    $,
+    ajax,
+    config,
+    mediator,
+    template,
+    SearchTool,
+    userPrefs,
+    weatherTemplate
+    ) {
+
+    var self           = null,
+        $weather       = null,
+        $holder        = null,
+        searchTool     = null,
+        prefName       = 'weather-location',
+        getGeoStates   = {
+            process: 'Getting location...',
+            error: 'Unable to get location...',
+            defaultmsg: 'Detect my location'
+        },
+        weatherApiUrl  = '/weather/city',
+        locationApiUrl = '/weather/locations?query=';
+
+    return {
+        init: function () {
+            self = this;
+
+            this.getDefaultLocation();
+        },
+
+        getGeoLocation: function () {
+            // This sends all geolocation data to fetchData when fetchData is expecting an Accuweather location
+            navigator.geolocation.getCurrentPosition(this.fetchData, this.geoLocationDisabled);
+        },
+
+        geoLocationDisabled: function () {
+            self.changeLocationOptionText('error');
+        },
+
+        detectPosition: function (e) {
+            e.preventDefault();
+
+            self.changeLocationOptionText('process');
+            self.getGeoLocation();
+        },
+
+        changeLocationOptionText: function (state) {
+            $('.js-detect-location').text(getGeoStates[state]);
+        },
+
+        /**
+         * Check if user has data in local storage.
+         * If yes return data from local storage else return default location data.
+         *
+         * @returns {object} geolocation - lat and long
+         */
+        getUserLocation: function () {
+            var prefs = userPrefs.get(prefName);
+
+            if (prefs && prefs.id) {
+                return prefs;
+            }
+        },
+
+        getWeatherData: function (url) {
+            return ajax({
+                url: url,
+                type: 'json',
+                method: 'get',
+                crossOrigin: true
+            });
+        },
+
+        /**
+         * Save user location into localStorage
+         */
+        saveUserLocation: function (location) {
+            userPrefs.set(prefName, {
+                'id': location.id,
+                'city': location.city
+            });
+        },
+
+        getDefaultLocation: function () {
+            var location = this.getUserLocation();
+
+            if (location) {
+                this.fetchData(location);
+            } else {
+                try {
+                    self.getWeatherData(weatherApiUrl + '.json').then(function (response) {
```

Comment:
Why create the anonymous function here? i.e.:
`self.getWeatherData(weatherApiUrl + '.json').then(self.fetchData)`
should suffice.